### PR TITLE
invoke FinishBinding by defer statement

### PR DESCRIPTION
**Pull request by gaorong**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
improve readability of bind function,  avoid making people confused about "Why is it reasonable to attempt FinishBinding if Bind returns an err?", such as  [here](https://github.com/kubernetes/kubernetes/pull/68444#discussion_r216475938) and [here](https://github.com/kubernetes/kubernetes/pull/68724#discussion_r221148003)

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
SGTM, as per the linked comments.
needs evaluation if the change of execution logic will break something.

/kind cleanup
/priority backlog
/ok-to-test

**Comment by gaorong** [comment]
> needs evaluation if the change of execution logic will break something.

IIUC, this won't break any rules, The side effect is it can imporve metric's accuracy below Bind funtion more or less :)

could you please help review? @bsalamat @k82cn

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -405,14 +405,16 @@ func (sched *Scheduler) assume(assumed *v1.Pod, host string) error {
 // bind binds a pod to a given node defined in a binding object.  We expect this to run asynchronously, so we
 // handle binding metrics internally.
 func (sched *Scheduler) bind(assumed *v1.Pod, b *v1.Binding) error {
+	defer func() {
```
[the remark this answers is not in the document]
Comment (krmayankk) [reply pkg/scheduler/scheduler.go]: i am not sure if calling FinishBinding after ForgetPod is OK. But i do find the logic and how its coded confusing. What advantage does it give to have bind, finishBinding be two parts

**Comment by k82cn** [comment]
Personally, I'd like to avoid such kind of refactor: we don-t have problem, and it's hard to evaluate impact because of race condition :).

**Comment by gaorong** [comment]
Thanks all of you, I totally understand what we concern about. this function is a key path to schedule pod, every change should take more attention. As this is a cleanup and not a must change, I'll close this PR.

**Comment by k82cn** [comment]
thanks very much for your understanding :)

**Pull request closed without merging**